docs/rfcs/RFC0008-OS-Packaging-Requirements.md

```diff
@@ -0,0 +1,313 @@
+| rocm-fft |  | rocFFT, hipFFT, hipFFTW |  |
+| rocm-math |  | Temporary catch all if libraries cannot fix circular dependencies by ROCm 8.0 |  |
+| rocm-blas | hipBLAS-common | rocBLAS, hipBLAS, hipBLASLt, hipSPARSELt |  |
+| rocm-sparse |  | rocSPARSE, hipSPARSE |  |
+| rocm-solver |  | rocSOLVER, hipSOLVER |  |
+| rocm-dnn |  | hipDNN, MIOpen |  |
+| rocm-rand |  | rocRAND, hipRAND |  |
+| rocm-ccl | rocPRIM, rocThrust, hipCUB | libhipcxx |  |
+| rocm-profiler |  | rocm-systems, rocm-compute, rocprofiler-sdk, tracer |  |
+| rocm-profiler-base |  | rocprofiler-sdk, tracer |  |
+| rocm-base |  | AMD-SMI, rocminfo, version (rocm-core) |  |
+| rocm-CK |  |  | CK |
+| rocm-debugger |  | rocgdb |  |
+| rocm-math-comon or rocm-math-dev |  |  | CK, rocWMMA, rocRoller, Tensile, Origami |
+| rocm-hipify |  | HIPIFY |  |
+| rocm-opencl |  | OpenCL |  |
+| rocm-decode |  | rocDecode |  |
+| rocm-jpeg |  | rocJPEG |  |
+| rocm-file |  | hipFile, rocFile (future addition) |  |
+| rocm-sysdeps |  | Bundled 3rd party dependencies (e.g. libdrm, libelf, numa, subset of libVA) |  |
+
+## Python Packaging Requirements
+
+### Standard pip WHeels
+
+Today's pip cannot handle cross-architecture extras without vitrual environments. The new packaging requirements will support the following options:
+
+```
+pip install rocm # GPU-agnostic
+pip install rocm[gfx908] # Single GPU architecture
+pip install rocm[gfx90a, gfx908] # Multiple GPU architectures (currently not supported)
+pip install rocm[all] # All GPU architectures
+```
+
+Engineering must validate the minimum Python version that supports bracket-extras.
+Dependencies should use `>=` (version X or newer) unless technical risk requires `==`.
+
+### WheelNext Variant Wheels
+
+WheelNext allows GPU detection via the amd-variant-provider which can be seen in the following examples:
+
+```
+uv pip install rocm
+uv pip install rocm[gfx90a, gfx908] # override default variant search
+uv pip install rocm[all]
+```
+
+TheRock must support building and publishing both:
+
+- Standard pip wheels
+- WheelNext variant wheels
+
+### Detection Requirements
+
+WheelNext must detect local GPUs via:
+
+- PCI bus scanning
+- ROCm driver presence
+- host-only variant
+
+## Versioning Requirements
+
+The following versioning format will be implemented as follows.
+
+Stable releases:
+
+```
+X.Y.Z
+```
+
+Prereleases:
+
+| Type of package | Version format | Version example |
+| :------------- | :------------- | :------------- |
+| Python | X.Y.ZrcN | 7.9.0rc0 |
+| Tarball | X.Y.ZrcN | 7.9.0rc0 |
+| Fedora (rpm) | X.Y.Z~rcN | 7.9.0~rc0 |
+| Debian (deb) | X.Y.Z~preN | 7.9.0~pre0 |
+
+Nightly Releases:
+
+| Type of package | Version format | Version example |
+| :------------- | :------------- | :------------- |
+| Python | X.Y.ZaYYYYMMDD | 7.10.0a20251006 |
+| Tarball | X.Y.ZaYYYYMMDD | 7.10.0a20251006 |
+| Fedora (rpm) | X.Y.Z~gitYYYYMMDD.${hash}-N | 7.10.0~git20251006.a961743280c591a1b8fe197ddf2928abd2f53a86-1 |
```

Comment:
git hash is not mandatory. Also from artifacts we wont be able to add module git hash in packages. Should remove that. 
rpm build tools mandates version in the format package_version-release_version .
expected rpm package name format : name-version-release.arch.rpm
So version should be 7.10.0-0a20251006  or 7.10.0a20251006-1

Would suggest to keep same format for Debian also. 

One more query. adding 'a' in version means its an alpha release. What should be the version when we move to ROCm8.0 and beyond? we still need 'a'